src/controller/CHIPDeviceController.cpp

```diff
@@ -629,6 +630,8 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, co
 CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, RendezvousParameters & params)
 {
     MATTER_TRACE_SCOPE("EstablishPASEConnection", "DeviceCommissioner");
+    MATTER_LOG_METRIC_BEGIN(PASESession);
```

Comment:
This is begin/end within a single method. This should be RAII instead and existing tracing already supports such scopes (which have guaranteed nesting).

Do we have explicit examples where guaranteed nesting is not applicable?

Reply:
You already have a trace scope above? Why are you creating a new one?

Reply:
Because tracing is different than metrics. Tracing is for instrumenting the code execution for developers (The Darwin ones are mapped to signposts now). Metrics is about gather stats for getting performance of the quality of the code in the field. Very different goals

Reply:
This was resolved offline